src/cfgparse.c

```diff
@@ -1192,8 +1192,12 @@ int cfg_parse_mailers(const char *file, int linenum, char **args, int kwm)
 			err_code |= ERR_ALERT | ERR_FATAL;
 			goto out;
 		}
-
+		
+#ifdef __linux__
+		if (proto->sock_prot != IPPROTO_TCP && proto->sock_prot != IPPROTO_MPTCP) {
+#else
 		if (proto->sock_prot != IPPROTO_TCP) {
+#endif
```

Comment:
What about including `haproxy/proto_tcp.h`, removing the `ifdef` and just do:

```c
if (proto->sock_prot != proto_tcpv4.sock_prot) {
```

See my previous comment: https://github.com/CrapsDorian/haproxy/pull/1#discussion_r1587652459

(with the current solution, you would have to define `IPPROTO_MPTCP` if it was not defined)